app/src/main/java/com/sample/appconnectsample/ImageCaptureActivity.java

```diff
@@ -0,0 +1,195 @@
+        protected void onPreExecute() {
+            progressDialog = new ProgressDialog(ImageCaptureActivity.this, ProgressDialog.STYLE_SPINNER);
+            progressDialog.setMessage(getString(R.string.image_capture_saving_message));
+            progressDialog.setCancelable(false);
+            progressDialog.show();
+        }
+
+        @Override
+        protected Void doInBackground(Bitmap... params) {
+            // *** AppConnect ***
+            // Each secondary thread must create its own datastore instance and
+            // dispose of it when done
+            Datastore datastore = null;
+
+            try {
+                datastore = DatastoreFactory.create();
+                Subject subject = datastore.getSubject(subjectID);
+
+                ByteArrayOutputStream stream = new ByteArrayOutputStream();
+                currentImage.compress(Bitmap.CompressFormat.JPEG, 70, stream);
+                byte[] byteArray = stream.toByteArray();
+
+                // the api call to babbage to actually save the data:
+                subject.collectData(byteArray, "sample metadata", "image/jpeg");
+            }
+            catch (Exception ex) {
+                exception = ex;
+                message = ex.getMessage();
+            }
+            finally {
+                if (datastore != null)
+                    datastore.dispose();
+            }
+            return null;
+        }
+
+        @Override
+        protected void onPostExecute(Void aVoid) {
+            progressDialog.cancel();
+
+            if (message != null) {
+                Log.e(TAG, message);
+                new AlertDialog.Builder(ImageCaptureActivity.this).
+                    setTitle(R.string.image_capture_error_title).
+                    setMessage(message).
+                    setPositiveButton(R.string.ok_button, null).
+                    show();
+            }
+            else if (exception != null) {
+                Log.e(TAG, "The submit task failed", exception);
+                new AlertDialog.Builder(ImageCaptureActivity.this).
+                    setTitle(R.string.image_capture_failed_title).
+                    setMessage(R.string.image_capture_failed_message).
+                    setPositiveButton(R.string.ok_button, null).
+                    show();
+            }
+            else {
+                new AlertDialog.Builder(ImageCaptureActivity.this).
+                    setCancelable(false).
+                    setTitle(R.string.image_capture_succeeded_title).
+                    setMessage(R.string.image_capture_succeeded_message).
+                    setPositiveButton(R.string.ok_button, new DialogInterface.OnClickListener() {
+                        @Override
+                        public void onClick(DialogInterface dialog, int which) {
+                            currentImageView.setImageBitmap(null);
+                            currentImage = null;
+
+                            //finish();
```

Comment:
nah, we stay on that activity after queueing upload, so the user could upload more if they like.